# Facebook likes en films
Data opschonen
Data:
- Aantal films
- Gemiddeld aantal likes per film
- Director met meeste likes
- Actor met meeste likes

Graphs:
- Movies per jaar
- Film tijd per jaar
- Imdb score vs facebook likes

In [141]:
# Bibliotheken
from datetime import datetime
import pandas as pd
import numpy as np
import scipy as sp
import matplotlib.pyplot as plt
from sklearn.decomposition import PCA
from sklearn.cluster import KMeans
from bokeh.plotting import *
from bokeh.layouts import row, widgetbox
from bokeh.models import CustomJS, Slider

bokeh.plotting.output_notebook()

%matplotlib inline

Loading BokehJS ...

In [4]:
movieData = pd.read_csv('movie_metadata.csv', sep=',')

movieData

color       director_name  num_critic_for_reviews  duration  \
0                Color       James Cameron                   723.0     178.0   
1                Color      Gore Verbinski                   302.0     169.0   
2                Color          Sam Mendes                   602.0     148.0   
3                Color   Christopher Nolan                   813.0     164.0   
4                  NaN         Doug Walker                     NaN       NaN   
5                Color      Andrew Stanton                   462.0     132.0   
6                Color           Sam Raimi                   392.0     156.0   
7                Color        Nathan Greno                   324.0     100.0   
8                Color         Joss Whedon                   635.0     141.0   
9                Color         David Yates                   375.0     153.0   
10               Color         Zack Snyder                   673.0     183.0   
11               Color        Bryan Singer                   434.0     169.0   
12               Color        Marc Forster                   403.0     106.0   
13               Color      Gore Verbinski                   313.0     151.0   
14               Color      Gore Verbinski                   450.0     150.0   
15               Color         Zack Snyder                   733.0     143.0   
16               Color      Andrew Adamson                   258.0     150.0   
17               Color         Joss Whedon                   703.0     173.0   
18               Color        Rob Marshall                   448.0     136.0   
19               Color    Barry Sonnenfeld                   451.0     106.0   
20               Color       Peter Jackson                   422.0     164.0   
21               Color           Marc Webb                   599.0     153.0   
22               Color        Ridley Scott                   343.0     156.0   
23               Color       Peter Jackson                   509.0     186.0   
24               Color         Chris Weitz                   251.0     113.0   
25               Color       Peter Jackson                   446.0     201.0   
26               Color       James Cameron                   315.0     194.0   
27               Color       Anthony Russo                   516.0     147.0   
28               Color          Peter Berg                   377.0     131.0   
29               Color     Colin Trevorrow                   644.0     124.0   
...                ...                 ...                     ...       ...   
5013             Color          Eric Eason                    28.0      79.0   
5014             Color            Uwe Boll                    58.0      80.0   
5015   Black and White   Richard Linklater                    61.0     100.0   
5016             Color     Joseph Mazzella                     NaN      90.0   
5017             Color        Travis Legge                     1.0      90.0   
5018             Color       Alex Kendrick                     5.0     120.0   
5019             Color       Marcus Nispel                    43.0      91.0   
5020               NaN     Brandon Landers                     NaN     143.0   
5021             Color         Jay Duplass                    51.0      85.0   
5022   Black and White          Jim Chuchu                     6.0      60.0   
5023             Color          Daryl Wein                    22.0      88.0   
5024             Color         Jason Trost                    42.0      78.0   
5025             Color         John Waters                    73.0     108.0   
5026             Color     Olivier Assayas                    81.0     110.0   
5027             Color        Jafar Panahi                    64.0      90.0   
5028   Black and White       Ivan Kavanagh                    12.0      83.0   
5029             Color    Kiyoshi Kurosawa                    78.0     111.0   
5030             Color        Tadeo Garcia                     NaN      84.0   
5031             

In [5]:
imdbExtraData = pd.read_csv('IMDB-Movie-Data.csv', sep=',')
imdbExtraData

Rank                                    Title  \
0       1                  Guardians of the Galaxy   
1       2                               Prometheus   
2       3                                    Split   
3       4                                     Sing   
4       5                            Suicide Squad   
5       6                           The Great Wall   
6       7                               La La Land   
7       8                                 Mindhorn   
8       9                       The Lost City of Z   
9      10                               Passengers   
10     11  Fantastic Beasts and Where to Find Them   
11     12                           Hidden Figures   
12     13                                Rogue One   
13     14                                    Moana   
14     15                                 Colossal   
15     16                  The Secret Life of Pets   
16     17                            Hacksaw Ridge   
17     18                             Jason Bourne   
18     19                                     Lion   
19     20                                  Arrival   
20     21                                     Gold   
21     22                    Manchester by the Sea   
22     23                           Hounds of Love   
23     24                                   Trolls   
24     25             Independence Day: Resurgence   
25     26                          Paris pieds nus   
26     27                  Bahubali: The Beginning   
27     28                               Dead Awake   
28     29                                 Bad Moms   
29     30                         Assassin's Creed   
..    ...                                      ...   
970   971                        Texas Chainsaw 3D   
971   972                                Disturbia   
972   973                             Rock of Ages   
973   974                                 Scream 4   
974   975                           Queen of Katwe   
975   976               My Big Fat Greek Wedding 2   
976   977                              Dark Places   
977   978                            Amateur Night   
978   979           It's Only the End of the World   
979   980                       The Skin I Live In   
980   981                     Miracles from Heaven   
981   982                                    Annie   
982   983                      Across the Universe   
983   984                            Let's Be Cops   
984   985                                      Max   
985   986                            Your Highness   
986   987                      Final Destination 5   
987   988                             Endless Love   
988   989                                  Martyrs   
989   990                                    Selma   
990   991           Underworld: Rise of the Lycans   
991   992                         Taare Zameen Par   
992   993                     Take Me Home Tonight   
993   994                 Resident Evil: Afterlife   
994   995                                Project X   
995   996                     Secret in Their Eyes   
996   997                          Hostel: Part II   
997   998                   Step Up 2: The Streets   
998   999                             Search Party   
999  1000                               Nine Lives   

                          Genre  \
0       Action,Adventure,Sci-Fi   
1      Adventure,Mystery,Sci-Fi   
2               Horror,Thriller   
3       Animation,Comedy,Family   
4      Action,Adventure,Fantasy   
5      Action,Adventure,Fantasy   
6            Comedy,Drama,Music   
7                        Comedy   
8    Action,Adventure,Biography   
9       Adventure,Drama,Romance   
10     Adventure,Family,Fantasy   
11      Biography,Drama,History   
12      Action,Adventure,Sci-Fi   
13   Animation,Adventure,Comedy   
14          Action,Comedy,Drama   
15   Animation,Adventure,Comedy   
16      Biography,Drama,History   
17              Action,Thr

## Data Analyze

Wij willen eerst kijken wat voor data wij allemaal hebben en hoe compleet deze data is:

In [6]:
print(movieData.columns)

Index(['color', 'director_name', 'num_critic_for_reviews', 'duration',
       'director_facebook_likes', 'actor_3_facebook_likes', 'actor_2_name',
       'actor_1_facebook_likes', 'gross', 'genres', 'actor_1_name',
       'movie_title', 'num_voted_users', 'cast_total_facebook_likes',
       'actor_3_name', 'facenumber_in_poster', 'plot_keywords',
       'movie_imdb_link', 'num_user_for_reviews', 'language', 'country',
       'content_rating', 'budget', 'title_year', 'actor_2_facebook_likes',
       'imdb_score', 'aspect_ratio', 'movie_facebook_likes'],
      dtype='object')


In [7]:
print(imdbExtraData.columns)

Index(['Rank', 'Title', 'Genre', 'Description', 'Director', 'Actors', 'Year',
       'Runtime (Minutes)', 'Rating', 'Votes', 'Revenue (Millions)',
       'Metascore'],
      dtype='object')


Graag willen wij ook even weten wat voor typen data wij hebben:

In [8]:
movieArray = movieData.infer_objects()

movieArray.dtypes

color                         object
director_name                 object
num_critic_for_reviews       float64
duration                     float64
director_facebook_likes      float64
actor_3_facebook_likes       float64
actor_2_name                  object
actor_1_facebook_likes       float64
gross                        float64
genres                        object
actor_1_name                  object
movie_title                   object
num_voted_users                int64
cast_total_facebook_likes      int64
actor_3_name                  object
facenumber_in_poster         float64
plot_keywords                 object
movie_imdb_link               object
num_user_for_reviews         float64
language                      object
country                       object
content_rating                object
budget                       float64
title_year                   float64
actor_2_facebook_likes       float64
imdb_score                   float64
aspect_ratio                 float64
m

In [9]:
imdbArray = imdbExtraData.infer_objects()

imdbArray.dtypes

Rank                    int64
Title                  object
Genre                  object
Description            object
Director               object
Actors                 object
Year                    int64
Runtime (Minutes)       int64
Rating                float64
Votes                   int64
Revenue (Millions)    float64
Metascore             float64
dtype: object

Wat er dan nog verkeerd aan onze Numpy Arrays is, is dat de String Lists (Bijvoorbeeld Genre's of Acteurs) geen lists zijn. 

Voordat wij dit kunnen fixen moeten wij eerst rows droppen die NaN zijn. 

Eerst willen wij weten hoeveel NaN values wij nou echt hebben.

In [10]:
print(pd.isnull(movieArray).sum())

color                         19
director_name                104
num_critic_for_reviews        50
duration                      15
director_facebook_likes      104
actor_3_facebook_likes        23
actor_2_name                  13
actor_1_facebook_likes         7
gross                        884
genres                         0
actor_1_name                   7
movie_title                    0
num_voted_users                0
cast_total_facebook_likes      0
actor_3_name                  23
facenumber_in_poster          13
plot_keywords                153
movie_imdb_link                0
num_user_for_reviews          21
language                      12
country                        5
content_rating               303
budget                       492
title_year                   108
actor_2_facebook_likes        13
imdb_score                     0
aspect_ratio                 329
movie_facebook_likes           0
dtype: int64


In [11]:
print(pd.isnull(imdbArray).sum())

Rank                    0
Title                   0
Genre                   0
Description             0
Director                0
Actors                  0
Year                    0
Runtime (Minutes)       0
Rating                  0
Votes                   0
Revenue (Millions)    128
Metascore              64
dtype: int64


Wat we best snel kunnen zien is dat de Gross Revenue vaak NaN is, in onze eerste dataset hebben wij van 884 films hun revenue gewoon helemaal niet (Dat is al bijna 1/5de van onze volledige dataset).

Dit kan later voor problemen zorgen wanneer wij machine learning gebruiken om te voorspellen hoeveel revenue een film krijgt gebaseerd op facebook likes van de acteurs/cast/directeurs.

Laten wij eerst de data een beetje opschonen, ten eerste willen wij de Strings die eigenlijk lijsten van acteurs/genres/etc zijn omzetten naar echte lijsten. Ook willen wij rows weggooien die data hebben die wij moeten hebben, deze data is: Revenue (Millions), Metascore, imdb_score, director_name, director_facebook_likes, actor_1_name, actor_1_facebook_likes, actor_2_name, actor_2_facebook_likes, actor_3_name, actor_3_facebook_likes, budget en gross.

Dit doe ik hier:

In [12]:
imdbArray = imdbArray.dropna(subset=['Revenue (Millions)'])
imdbArray = imdbArray.dropna(subset=['Metascore'])

movieArray = movieArray.dropna(subset=['imdb_score'])
movieArray = movieArray.dropna(subset=['director_name'])
movieArray = movieArray.dropna(subset=['director_facebook_likes'])
movieArray = movieArray.dropna(subset=['actor_1_name'])
movieArray = movieArray.dropna(subset=['actor_1_facebook_likes'])
movieArray = movieArray.dropna(subset=['actor_2_name'])
movieArray = movieArray.dropna(subset=['actor_2_facebook_likes'])
movieArray = movieArray.dropna(subset=['actor_3_name'])
movieArray = movieArray.dropna(subset=['actor_3_facebook_likes'])
movieArray = movieArray.dropna(subset=['budget'])
movieArray = movieArray.dropna(subset=['gross'])

#Na het verwijderen van NaN values moet de index ook nog gereset worden van de Arrays, anders kan dat snel voor errors zorgen

movieArray.reset_index(drop=True, inplace=True)
imdbArray.reset_index(drop=True, inplace=True)

In [13]:
print(pd.isnull(movieArray).sum())

color                         2
director_name                 0
num_critic_for_reviews        1
duration                      1
director_facebook_likes       0
actor_3_facebook_likes        0
actor_2_name                  0
actor_1_facebook_likes        0
gross                         0
genres                        0
actor_1_name                  0
movie_title                   0
num_voted_users               0
cast_total_facebook_likes     0
actor_3_name                  0
facenumber_in_poster          6
plot_keywords                30
movie_imdb_link               0
num_user_for_reviews          0
language                      3
country                       0
content_rating               47
budget                        0
title_year                    0
actor_2_facebook_likes        0
imdb_score                    0
aspect_ratio                 73
movie_facebook_likes          0
dtype: int64


In [14]:
print(pd.isnull(imdbArray).sum())

Rank                  0
Title                 0
Genre                 0
Description           0
Director              0
Actors                0
Year                  0
Runtime (Minutes)     0
Rating                0
Votes                 0
Revenue (Millions)    0
Metascore             0
dtype: int64


Nu hebben wij dus geen rows meer met NaN data, dit kan wel zorgen voor minder correcte data. Maar sinds wij 2 zeer grote Datasets hebben zou dit geen probleem horen te zijn.

Nu er geen NaN data meer is kan ik met een lambda en de split functie alle columns die een String list in een String hebben omzetten naar lists.

In [15]:
movieArray['genres'] = movieArray['genres'].apply(lambda x : x.split('|'))

movieArray['genres']

0                    [Action, Adventure, Fantasy, Sci-Fi]
1                            [Action, Adventure, Fantasy]
2                           [Action, Adventure, Thriller]
3                                      [Action, Thriller]
4                             [Action, Adventure, Sci-Fi]
5                            [Action, Adventure, Romance]
6       [Adventure, Animation, Comedy, Family, Fantasy...
7                             [Action, Adventure, Sci-Fi]
8                   [Adventure, Family, Fantasy, Mystery]
9                             [Action, Adventure, Sci-Fi]
10                            [Action, Adventure, Sci-Fi]
11                                    [Action, Adventure]
12                           [Action, Adventure, Fantasy]
13                           [Action, Adventure, Western]
14                   [Action, Adventure, Fantasy, Sci-Fi]
15                   [Action, Adventure, Family, Fantasy]
16                            [Action, Adventure, Sci-Fi]
17            

In [16]:
#Split geeft errors bij een NaN value
movieArray = movieArray.dropna(subset=['plot_keywords'])
movieArray['plot_keywords'] = movieArray['plot_keywords'].apply(lambda x : x.split('|'))
movieArray['plot_keywords']

0            [avatar, future, marine, native, paraplegic]
1       [goddess, marriage ceremony, marriage proposal...
2               [bomb, espionage, sequel, spy, terrorist]
3       [deception, imprisonment, lawlessness, police ...
4       [alien, american civil war, male nipple, mars,...
5         [sandman, spider man, symbiote, venom, villain]
6       [17th century, based on fairy tale, disney, fl...
7       [artificial intelligence, based on comic book,...
8                  [blood, book, love, potion, professor]
9       [based on comic book, batman, sequel to a rebo...
10      [crystal, epic, lex luthor, lois lane, return ...
11      [action hero, attempted rape, bond girl, offic...
12      [box office hit, giant squid, heart, liar's di...
13            [horse, outlaw, texas, texas ranger, train]
14      [based on comic book, british actor playing am...
15      [brother brother relationship, brother sister ...
16      [alien invasion, assassin, battle, iron man, s...
17        [bla

In [17]:
imdbArray['Genre'] = imdbArray['Genre'].apply(lambda x : x.split(','))
imdbArray['Genre']

0         [Action, Adventure, Sci-Fi]
1        [Adventure, Mystery, Sci-Fi]
2                  [Horror, Thriller]
3         [Animation, Comedy, Family]
4        [Action, Adventure, Fantasy]
5        [Action, Adventure, Fantasy]
6              [Comedy, Drama, Music]
7      [Action, Adventure, Biography]
8         [Adventure, Drama, Romance]
9        [Adventure, Family, Fantasy]
10        [Biography, Drama, History]
11        [Action, Adventure, Sci-Fi]
12     [Animation, Adventure, Comedy]
13            [Action, Comedy, Drama]
14     [Animation, Adventure, Comedy]
15        [Biography, Drama, History]
16                 [Action, Thriller]
17                 [Biography, Drama]
18           [Drama, Mystery, Sci-Fi]
19       [Adventure, Drama, Thriller]
20                            [Drama]
21     [Animation, Adventure, Comedy]
22        [Action, Adventure, Sci-Fi]
23                           [Comedy]
24         [Action, Adventure, Drama]
25                           [Comedy]
26          

Voor onze eerste dataset hebben wij ook de column Color niet nodig, wij gooien deze dus weg

In [18]:
movieArray = movieArray.drop(columns='color')
movieArray

director_name  num_critic_for_reviews  duration  \
0          James Cameron                   723.0     178.0   
1         Gore Verbinski                   302.0     169.0   
2             Sam Mendes                   602.0     148.0   
3      Christopher Nolan                   813.0     164.0   
4         Andrew Stanton                   462.0     132.0   
5              Sam Raimi                   392.0     156.0   
6           Nathan Greno                   324.0     100.0   
7            Joss Whedon                   635.0     141.0   
8            David Yates                   375.0     153.0   
9            Zack Snyder                   673.0     183.0   
10          Bryan Singer                   434.0     169.0   
11          Marc Forster                   403.0     106.0   
12        Gore Verbinski                   313.0     151.0   
13        Gore Verbinski                   450.0     150.0   
14           Zack Snyder                   733.0     143.0   
15        Andrew Adamson                   258.0     150.0   
16           Joss Whedon                   703.0     173.0   
17          Rob Marshall                   448.0     136.0   
18      Barry Sonnenfeld                   451.0     106.0   
19         Peter Jackson                   422.0     164.0   
20             Marc Webb                   599.0     153.0   
21          Ridley Scott                   343.0     156.0   
22         Peter Jackson                   509.0     186.0   
23           Chris Weitz                   251.0     113.0   
24         Peter Jackson                   446.0     201.0   
25         James Cameron                   315.0     194.0   
26         Anthony Russo                   516.0     147.0   
27            Peter Berg                   377.0     131.0   
28       Colin Trevorrow                   644.0     124.0   
29            Sam Mendes                   750.0     143.0   
...                  ...                     ...       ...   
3849       Alex Kendrick                    31.0     111.0   
3850        Travis Cluff                   159.0      81.0   
3851   Harry F. Millarde                     1.0     110.0   
3852     Robert Townsend                    21.0      81.0   
3853       Larry Blamire                    88.0      90.0   
3854           E.L. Katz                   193.0      88.0   
3855      Dennis Iliadis                   241.0     114.0   
3856    Darren Aronofsky                   138.0      84.0   
3857     Myles Berkowitz                    32.0      87.0   
3858     Morgan Spurlock                   193.0     100.0   
3859       Brandon Trost                    66.0      82.0   
3860        Joe Swanberg                    65.0      82.0   
3861        Edward Burns                    36.0      98.0   
3862         Lena Dunham                   113.0      98.0   
3863  David Gordon Green                    75.0      90.0   
3864        Kevin Jordan                    21.0      90.0   
3866         Kevin Smith                   136.0     102.0   
3867         Neil LaBute                    80.0      97.0   
3868          David Ayer                   233.0     109.0   
3869   Richard Linklater                    61.0     100.0   
3870         Jay Duplass                    51.0      85.0   
3872         John Waters                    73.0     108.0   
3873     Olivier Assayas                    81.0     110.0   
3874        Jafar Panahi                    64.0      90.0   
3875    Kiyoshi Kurosawa                    78.0     111.0   
3876       Shane Carruth                   143.0      77.0   
3877    Neill Dela Llana                    35.0      80.0   
3878    Robert Rodriguez                    56.0      81.0   
3879        Edward Burns                    14.0      95.0   
3880            Jon Gunn                    43.0      90.0   

      director_facebook_likes  actor_3_facebook_likes          actor_2_name  \
0                         0.0                   855.0      Joel David Moore   
1       

Wat wij dus willen weten is de gross revenue gebaseerd op IMDB score en gebaseerd op Facebook likes. Wij mergen dus eerst onze datasets samen zodat wij een globale dataset hebben die wij hiervoor kunnen gebruiken:

In [19]:
arraysToKeepMovieArray = ['movie_title', 'imdb_score', 'gross', 'director_facebook_likes', 'actor_1_facebook_likes', 'actor_2_facebook_likes', 'actor_3_facebook_likes']

ndf = movieArray[arraysToKeepMovieArray]

#Het decimaalpunt stond verkeerd in deze
ndf['imdb_score'] = ndf.imdb_score.apply(lambda x: x * 10)

C:\Users\atill\Anaconda3\lib\site-packages\ipykernel_launcher.py:6: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  


In [21]:
arraysToKeepImdbArray = ['movie_title', 'imdb_score', 'gross']

imdbArray = imdbArray.rename(columns={'Title' : 'movie_title', 'Metascore' : 'imdb_score', 'Revenue (Millions)' : 'gross'})

ndf2 = imdbArray[arraysToKeepImdbArray]

imdbMergedArray = ndf.merge(ndf2, how='left')

imdbMergedArray = imdbMergedArray.reset_index()

imdbMergedArray = imdbMergedArray.set_index('index')

imdbMergedArray

movie_title  imdb_score        gross  \
index                                                                          
0                                           Avatar         79.0  760505847.0   
1         Pirates of the Caribbean: At World's End         71.0  309404152.0   
2                                          Spectre         68.0  200074175.0   
3                            The Dark Knight Rises         85.0  448130642.0   
4                                      John Carter         66.0   73058679.0   
5                                     Spider-Man 3         62.0  336530303.0   
6                                          Tangled         78.0  200807262.0   
7                          Avengers: Age of Ultron         75.0  458991599.0   
8           Harry Potter and the Half-Blood Prince         75.0  301956980.0   
9               Batman v Superman: Dawn of Justice         69.0  330249062.0   
10                                Superman Returns         61.0  200069408.0   
11                               Quantum of Solace         67.0  168368427.0   
12      Pirates of the Caribbean: Dead Man's Chest         73.0  423032628.0   
13                                 The Lone Ranger         65.0   89289910.0   
14                                    Man of Steel         72.0  291021565.0   
15        The Chronicles of Narnia: Prince Caspian         66.0  141614023.0   
16                                    The Avengers         81.0  623279547.0   
17     Pirates of the Caribbean: On Stranger Tides         67.0  241063875.0   
18                                  Men in Black 3         68.0  179020854.0   
19       The Hobbit: The Battle of the Five Armies         75.0  255108370.0   
20                          The Amazing Spider-Man         70.0  262030663.0   
21                                      Robin Hood         67.0  105219735.0   
22             The Hobbit: The Desolation of Smaug         79.0  258355354.0   
23                              The Golden Compass         61.0   70083519.0   
24                                       King Kong         72.0  218051260.0   
25                                         Titanic         77.0  658672302.0   
26                      Captain America: Civil War         82.0  407197282.0   
27                                      Battleship         59.0   65173160.0   
28                                  Jurassic World         70.0  652177271.0   
29                                         Skyfall         78.0  304360277.0   
...                                             ...         ...          ...   
3821                             Facing the Giants         67.0   10174663.0   
3822                                   The Gallows         42.0   22757819.0   
3823                Over the Hill to the Poorhouse         48.0    3000000.0   
3824                             Hollywood Shuffle         70.0    5228617.0   
3825                  The Lost Skeleton of Cadavra         70.0     110536.0   
3826                                 Cheap Thrills         68.0      59379.0   
3827                    The Last House on the Left         66.0   32721635.0   
3828                                            Pi         75.0    3216970.0   
3829                                      20 Dates         53.0     536767.0   
3830                                 Super Size Me         73.0   11529368.0   
3831                                        The FP         56.0      40557.0   
3832                               Happy Christmas         56.0      30084.0   
3833                         The Brothers McMullen         66.0   10246600.0   
3834                                Tiny Furniture         63.0     389804.0   
3835                             George Washington         75.0     241816.0   
3836                   Smiling Fish & Goat on Fire         76.0     277233.0   
3837                                        Clerks         78.0    3151130.0   
3838                         In the Comp

Wij hebben nu een gemergede Array die de movie_title, imdb_score, gross revenue en alle likes heeft. Dit maakt het zeer makkelijk om alles te plotten.

### Data uitleg

Wat wij dus willen bekijken is hoe belangrijk het aantal facebook likes van een directeur/acteur is voor een film, dit doen wij dus door de directeur/acteur facebook likes met de IMDB scores te vergelijken, en door de facebook likes met de revenue te vergelijken.

Eerst kijken wij hoe de scores zijn verdeeld, hieronder zul je de normaalverdeling zien.

In [ ]:
mu_cijfers = imdbMergedArray.imdb_score.mean()
sigma_cijfers = imdbMergedArray.imdb_score.std()
normal_cijfers = sp.stats.norm(mu_cijfers, sigma_cijfers)

In [ ]:
x = np.linspace(20,100,2000)
y = normal_cijfers.pdf(x)

plt.plot(x,y)
plt.fill_between(x,y, where=sp.logical_and(x>60, x<85))
plt.show()

## Gemiddeldes

In [ ]:
averageLikes = movieData["movie_facebook_likes"].mean()
print("Gemidelde aantal likes per film is " + str(averageLikes))

In [ ]:
movieData["movie_facebook_likes"].std()

## Meeste likes 

In [ ]:
def getTotalLikesByColumn(column):
    groupedFacebookLikes = movieData.groupby(column)["movie_facebook_likes"].sum()
    key = groupedFacebookLikes.argmax()
    maxSum = groupedFacebookLikes.max()
    return str(key) + " heeft totaal "+ str(maxSum) + " likes";

In [ ]:
print("De director met de meeste likes:")
print(getTotalLikesByColumn("director_name"))

In [ ]:
print("De hoofd acteur met de meeste likes:")
print(getTotalLikesByColumn("actor_1_name"))

In [ ]:
print("De film met de meeste likes:")
print(getTotalLikesByColumn("movie_title"))

## Minste likes

In [ ]:
def getMinLikesByColumn(column):
    groupedFacebookLikes = movieData.groupby(column)["movie_facebook_likes"].sum()
    key = groupedFacebookLikes.argmin()
    maxSum = groupedFacebookLikes.min()
    return str(key) + " heeft totaal "+ str(maxSum) + " likes";

In [ ]:
print("De director met de minste likes:")
print(getMinLikesByColumn("director_name"))

In [ ]:
print("De hoofd acteur met de minste likes:")
print(getMinLikesByColumn("actor_1_name"))

In [ ]:
print("De film met de minste likes:")
print(getMinLikesByColumn("movie_title"))

## Graphs

In [ ]:
movieYears = movieData["title_year"]
movieYears = movieYears[movieYears.notnull()]

In [ ]:
movieYearLabels,movieYearCounts = np.unique(movieYears,return_counts=True)
y_pos = np.arange(len(movieYearLabels))

plt.plot(movieYearLabels, movieYearCounts)
plt.ylabel("Aantal")
plt.xlabel("Jaar")
plt.title("Aantal films per jaar")
plt.show()

In [ ]:
movieDurations = movieData.groupby("title_year")["duration"]
averageMovieTime = movieDurations.mean()

In [ ]:
averageMovieTime.plot()
plt.ylabel("tijd (minuten)")
plt.xlabel("Jaar")
plt.title("Gemiddelde film lengte per jaar")
plt.show()

In [ ]:
plt.plot(movieData["budget"], movieData["cast_total_facebook_likes"])
plt.ylabel("Total cast facebook likes")
plt.xlabel("Budget")
print("Facebook likes vs budget")
plt.show()

In [221]:
columnsToKeepImdbRevenue = ['imdb_score', 'gross']

imdbRevenue = imdbMergedArray[columnsToKeepImdbRevenue]

x_cluster = imdbRevenue['imdb_score']
y_cluster = imdbRevenue['gross']

kmeans = KMeans(n_clusters=2, random_state=0).fit(imdbRevenue)
centroids = kmeans.cluster_centers_

colors = np.array([x for x in ('#00f', '#0f0', '#f00', '#0ff', '#f0f', '#ff0')])

y_pred = kmeans.labels_.astype(np.int)

p = figure(x_axis_label='IMDB Score', y_axis_label='Gross Revenue')

p.scatter(x_cluster, y_cluster, fill_alpha=0.2, color=colors[y_pred].tolist())

for idx, centroid in enumerate(centroids):
    p.circle_cross(centroids[:,0], centroids[:,1], line_color='black', fill_color='blue', fill_alpha=0.3, size=30)

show(p)